transformer_lens/HookedSAETransformer.py

```diff
@@ -0,0 +1,284 @@
+                remove_batch_dim=remove_batch_dim,
+                **kwargs,
+            )
+
+    def run_with_hooks_with_saes(
+        self,
+        *model_args,
+        saes: Union[HookedSAE, List[HookedSAE]] = [],
+        reset_saes_end: bool = True,
+        fwd_hooks: List[Tuple[Union[str, Callable], Callable]] = [],
+        bwd_hooks: List[Tuple[Union[str, Callable], Callable]] = [],
+        reset_hooks_end=True,
+        clear_contexts=False,
+        **model_kwargs,
+    ):
+        """Wrapper around 'run_with_hooks' in HookedTransformer.
+
+        Attaches the given SAEs to the model before running the model with hooks and then removes them.
+        By default, will reset all SAEs to original state after.
+
+        Args:
+            *model_args: Positional arguments for the model forward pass
+            act_names: (Union[HookedSAE, List[HookedSAE]]) The SAEs to be attached for this forward pass
+            reset_saes_end: (bool) If True, all SAEs added during this run are removed at the end, and previously attached SAEs are restored to their original state. (default: True)
+            fwd_hooks: (List[Tuple[Union[str, Callable], Callable]]) List of forward hooks to apply
+            bwd_hooks: (List[Tuple[Union[str, Callable], Callable]]) List of backward hooks to apply
+            reset_hooks_end: (bool) Whether to reset the hooks at the end of the forward pass (default: True)
+            clear_contexts: (bool) Whether to clear the contexts at the end of the forward pass (default: False)
+            **model_kwargs: Keyword arguments for the model forward pass
+        """
+        with self.saes(saes=saes, reset_saes_end=reset_saes_end):
+            return self.run_with_hooks(
+                *model_args,
+                fwd_hooks=fwd_hooks,
+                bwd_hooks=bwd_hooks,
+                reset_hooks_end=reset_hooks_end,
+                clear_contexts=clear_contexts,
+                **model_kwargs,
+            )
+
+    @contextmanager
+    def saes(
+        self,
+        saes: Union[HookedSAE, List[HookedSAE]] = [],
+        reset_saes_end: bool = True,
+    ):
+        """
+        A context manager for adding temporary SAEs to the model.
+        See HookedTransformer.hooks for a similar context manager for hooks.
+        By default will keep track of previously attached SAEs, and restore them when the context manager exits.
+
+        Args:
+            saes (Union[HookedSAE, List[HookedSAE]]): SAEs to be attached.
+            reset_saes_end (bool): If True, removes all SAEs added by this context manager when the context manager exits, returning previously attached SAEs to their original state.
+
+        Example:
+            ```python
+            hooked_saes = [HookedSAE(...), HookedSAE(...)]
+            with model.saes(saes=hooked_saes):
+                spliced_loss = model(text, return_type="loss")
+            ```
+        """
+        act_names_to_reset = []
```

Comment:
```suggestion
    @contextmanager
    def saes(
        self,
        saes: Union[HookedSAE, List[HookedSAE]] = [],
        reset_saes_end: bool = True,
    ):
        """
        A context manager for adding temporary SAEs to the model.
        Attaches specified SAEs temporarily, and restores the previous state at exit.
        
        Args:
            saes (Union[HookedSAE, List[HookedSAE]]): The SAEs to attach temporarily.
            reset_saes_end (bool): If True, restores the SAEs' original state after exiting context.
        
        Example:
        ```python
        hooked_saes = [HookedSAE(...), HookedSAE(...)]
        with model.saes(saes=hooked_saes):
            spliced_loss = model(text, return_type="loss")
        ```
        """
        act_names_to_reset = []
```

Reply:
@ckkissane this is where the error is -- maybe either tab/space spacing is wrong, or it hates the \n here?

Reply:
For some reason the ```python block was causing issues, so I changed it to .. code-block:: python